src/status_im/chat/events/receive_message.cljs

```diff
@@ -16,30 +17,49 @@
+      ;; note that `request-command-message-data` implicitly wait till jail is ready and
+      ;; calls are made only after that
+      (commands-events/request-command-message-data
+       db message
+       {:data-type             :short-preview
+        :proceed-event-creator (fn [short-preview]
+                                 [:request-command-message-data
+                                  message
+                                  {:data-type             :preview
+                                   :proceed-event-creator (fn [preview]
+                                                            [::received-message
+                                                             (update message :content merge
+                                                                     {:short-preview short-preview
+                                                                      :preview       preview})])}])})
+      ;; regular non command message, we can add it right away
+      (message-model/receive message cofx))))
+
+(defn add [[messages] {:keys [db] :as cofx}]
+  (reduce (fn [fx message]
+            (let [temp-cofx (handlers-macro/update-db cofx fx)]
+              (handlers-macro/safe-merge
+               fx
+               (add-one-message [message] temp-cofx))))
+          {:db db}
+          messages))
+
 (handlers/register-handler-fx
  :chat-received-message/add
  message-model/receive-interceptors
- (fn [{:keys [db] :as cofx} [{:keys [content] :as message}]]
-   (when (message-model/add-to-chat? cofx message)
-     (if (:command content)
-        ;; we are dealing with received command message, we can't add it right away,
-        ;; we first need to fetch short-preview + preview and add it only after we already have those.
-        ;; note that `request-command-message-data` implicitly wait till jail is ready and
-        ;; calls are made only after that
-       (commands-events/request-command-message-data
-        db message
-        {:data-type             :short-preview
-         :proceed-event-creator (fn [short-preview]
-                                  [:request-command-message-data
-                                   message
-                                   {:data-type             :preview
-                                    :proceed-event-creator (fn [preview]
-                                                             [::received-message
-                                                              (update message :content merge
-                                                                      {:short-preview short-preview
-                                                                       :preview       preview})])}])})
-        ;; regular non command message, we can add it right away
-       (message-model/receive message cofx)))))
+ (fn [cofx args]
+   (if (vector? (first args))
+     (add args cofx)
+     (add-one-message args cofx))))
```

Comment:
add-message

Reply:
same

Reply:
both names are too un-conventionnal, add-message and add-messages has clearer intent